netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -248,7 +252,16 @@ public void operationComplete(ChannelFuture future) throws Exception {
       keepAliveManager.onTransportStarted();
     }
 
-    return null;
+    return new Runnable() {
+      @Override
+      public void run() {
+        try {
+          listenerDone.sync();
```

Comment:
We can't block like this without risking deadlock. Let's look for an alternative.